### PR TITLE
ccip-scripts job should upload output to s3

### PR DESCRIPTION
**Pull request by scheibinger**
## Motivation
ccip-scripts deploy job produces the output file which is required to run the tests.

## Solution
* Have ccip-scripts job to upload contracts and jobs output data to s3 so they can be further used in CI or CLI for running tests
  * Sending file over to S3 requires enabling egress traffic
  * job will use wrapper script which include s3 upload command
* Refactored configuration uptions to have parent ccipScriptsDeployment property

Depends on: 
* https://github.com/smartcontractkit/ccip-scripts/pull/143
* https://github.com/smartcontractkit/ccip-scripts/pull/142

### CODE REVIEW

File: charts/chainlink-cluster/templates/ccip-scripts-job.yaml [hunk 1]
```diff
@@ -43,11 +45,11 @@ spec:
             {{- end }}
       containers:
         - name: ccip-scripts
-          image: {{.Values.ccip.ccipScriptsImage}}
+          image: {{.Values.ccip.ccipScriptsDeployment.image}}
           env:
             - name: CONFIG_JSON_PATH
               value: /data/config.json
-          command: [ "pilot.test", "-test.v" ]
+          command: [ "./run.sh", "{{.Values.ccip.ccipScriptsDeployment.outputBucketName}}" ]
```
Comment (chainchad) [inline charts/chainlink-cluster/templates/ccip-scripts-job.yaml]: Want to confirm that this script is in the Docker image?

File: .tool-versions [hunk 2]
```diff
@@ -1,4 +1,4 @@
-golang 1.21.7
+golang 1.21.8
```
Comment (chainchad) [inline .tool-versions]: Does this bump have any implications for the repo as a whole?

File: charts/chainlink-cluster/templates/ccip-scripts-job.yaml (hunk 1 above)
Comment (scheibinger) [reply charts/chainlink-cluster/templates/ccip-scripts-job.yaml]: yup it's here: https://github.com/smartcontractkit/ccip-scripts/pull/143

File: .tool-versions (hunk 2 above)
Comment (scheibinger) [reply .tool-versions]: I made a separate PR to bump in repo as well, I found without this the `devspace build` and goreleaser cmd particularly was broken

**Review by chainchad — APPROVED** [review]

**Pull request merged**